tests/unittests/test_handler/test_schema.py

```diff
@@ -507,9 +551,76 @@ def _get_schema_doc_examples():
 class TestSchemaDocExamples:
     schema = get_schema()
 
-    @pytest.mark.parametrize("example_path", _get_schema_doc_examples())
+    @pytest.mark.parametrize("example_path", _get_meta_doc_examples())
     @skipUnlessJsonSchema()
     def test_schema_doc_examples(self, example_path):
         validate_cloudconfig_file(str(example_path), self.schema)
 
+
+class TestStrictMetaschema:
+    '''Validate that schemas follow a stricter metaschema definition than
+    the default. This disallows arbitrary key/value pairs.
+    '''
+    def test_modules(self):
+        '''Validate all modules with a stricter metaschema'''
+        for (name, value) in get_schemas().items():
+            if value:
+                self.validate_cloudconfig_schema_strict(value, name)
+            else:
+                logging.warning(
+                    "module %s has no schema definition", name)
+
+    def validate_cloudconfig_schema_strict(self, schema: dict, name: str):
+        """Validate schema definition against strict metaschema.
+
+        @param schema: jsonschema dict describing the supported schema
+           definition for the cloud config module (config.cc_*)
+        @param name: for error handling
+
+        This is a modified version of the validation function in schema.py
+        """
+        try:
+            from jsonschema import Draft4Validator, FormatChecker
+            from jsonschema.validators import create, extend
+        except ImportError:
+            logging.debug(
+                'Ignoring schema validation. python-jsonschema is not present')
+            return
+
+        # Allow for bytes to be presented as an acceptable valid value for
+        # string type jsonschema attributes in cloud-init's schema.
+        # This allows #cloud-config to provide valid yaml
+        # "content: !!binary | ..."
+        if hasattr(Draft4Validator, 'TYPE_CHECKER'):  # jsonschema 3.0+
+            type_checker = Draft4Validator.TYPE_CHECKER.redefine(
+                'string', is_schema_byte_string)
+            cloudinitValidator = extend(
+                Draft4Validator, type_checker=type_checker)
+        else:  # jsonschema 2.6 workaround
+            types = Draft4Validator.DEFAULT_TYPES
+            # Allow bytes as well as string (and disable a spurious unsupported
+            # assignment-operation pylint warning which appears because this
+            # code path isn't written against the latest jsonschema).
+            types['string'] = (str, bytes)  # pylint: disable=E1137
+            cloudinitValidator = create(
+                meta_schema=Draft4Validator.META_SCHEMA,
+                validators=Draft4Validator.VALIDATORS,
+                version="draft4",
+                default_types=types)
+
+        mymeta = cloudinitValidator.META_SCHEMA
+
+        # this disables bottom-level keys
```

Comment:
This line is basically the only difference between an otherwise identical function in `schema.py`. I'm leaning towards adding a flag to the function signature in schema.py to set additionalProperties=False, which would reduce this diff by about 70 LOC.